sympy/core/basic.py

```diff
@@ -1077,6 +1077,151 @@ def _eval_subs(self, old, new):
+
+        ``mapdict``: In case a transformation contains more than one new
+        variable, mapdict must be provided to tell SymPy which variable
+        should be considered for substitution. For eg:
+
+        If you want to substitute for `x = at^2` where `t` is the new
+        independent variable, `subsdict` must be {x: a*t**2} and mapdict
+        must be {x: t}. It is ``None`` by default.
+
+        Returns
+        =======
+
+        The transformed differential equation
+
+        Examples
+        ========
+
+        Let us see how ``dsubs`` works using an example of homogeneous
+        ODEs which can be solved using the substitution `v = \frac{y}{x}`.
+
+        >>> from sympy import symbols, Function, dsolve
+        >>> x, t = symbols('x t')
+        >>> y, v = symbols('y v', cls=Function)
+        >>> eq = y(x).diff(x) - (x**2 + y(x)**2)/(x*y(x))
+
+        Transform the homogeneous equation to a linear equation
+
+        >>> homeq = eq.dsubs({x:t, y(x): t*v(t)}).subs(t, x).simplify()
+        >>> homeq
+        x*Derivative(v(x), x) - 1/v(x)
+
+        Find the solutions for the transformed equation
+
+        >>> sols = dsolve(homeq)
+        >>> sols
+        [Eq(v(x), -sqrt(C1 + 2*log(x))), Eq(v(x), sqrt(C1 + 2*log(x)))]
+
+        Re-substitute `v = y x` to get the solutions to the
+        original equation
+
+        >>> sols = list(map(lambda sol:sol.subs(v(x), x*y(x)), sols))
+        >>> sols
+        [Eq(x*y(x), -sqrt(C1 + 2*log(x))), Eq(x*y(x), sqrt(C1 + 2*log(x)))]
+
+        Let us look at another example - The 2nd order Cauchy Euler equation.
+
+        >>> x, a, b, t = symbols('x a b t')
+        >>> y, phi = symbols('y phi', cls=Function)
+        >>> eq = x**2*y(x).diff(x, 2) + a*x*y(x).diff(x) + b*y(x)
+
+        Transform the equation using the substitutions `x = e^t`
+        and `y(x) = \phi(t)`
+
+        >>> from sympy import exp, log, sqrt, logcombine
+        >>> eqtrans = eq.dsubs({x: exp(t), y(x): phi(t)}).simplify()
+        >>> eqtrans
+        a*Derivative(phi(t), t) + b*phi(t) - Derivative(phi(t), t) + Derivative(phi(t), (t, 2))
+
+        The equation is now transformed to a constant-coefficient equation.
+
+        >>> sol = dsolve(eqtrans)
+        >>> sol
+        Eq(phi(t), C1*exp(t*(-a - sqrt(a**2 - 2*a - 4*b + 1) + 1)/2) + C2*exp(t*(-a + sqrt(a**2 - 2*a - 4*b + 1) + 1)/2))
+
+        Re-substitute for `\phi(t)` and `t` to get the solution for
+        the original equation.
+
+        >>> sol = logcombine(sol.subs(phi(t), y(x)).subs(t, log(x)), force=True)
+        >>> sol
+        Eq(y(x), C1*x**(-a/2 - sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2) + C2*x**(-a/2 + sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2))
+
+        If an equation contains multiple independent variables, one or more
+        of them can also be substituted. Note that this should not be confused
+        with PDE substitution. This is still a substitution in an ODE with
+        different independent variables.
```

Comment:
This doesn't really make any sense to me. An ODE can only have one independent variable.

Reply:
Its like a sum of different ODEs. The substitution must be possible in such cases as well. This works even in `dchange`.

Reply:
Is this fine or should I remove it from the docstring?